scripts/result_listing/compare_results.py

```diff
@@ -0,0 +1,57 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import argparse
+import json
+import sys
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Compares two CodeChecker results. The results should be "
+                    "generated by 'CodeChecker cmd results' command.")
+
+    parser.add_argument(
+        "result1",
+        type=argparse.FileType("r"),
+        help="Path of the first result.")
+
+    parser.add_argument(
+        "result2",
+        type=argparse.FileType("r"),
+        help="Path of the second result.")
+
+    args = parser.parse_args()
+
+    result1 = json.load(args.result1)
+    result2 = json.load(args.result2)
+
+    result1.sort(key=lambda report: report['bugHash'] + str(report['reportId']))
+    result2.sort(key=lambda report: report['bugHash'] + str(report['reportId']))
+
+    found_mismatch = False
+
+    for report in result1:
+        if report not in result2:
+            print("Report not found in result2:")
+            print(json.dumps(report, indent=4))
+            found_mismatch = True
+
+    for report in result2:
+        if report not in result1:
+            print("Report not found in result1:")
+            print(json.dumps(report, indent=4))
+            found_mismatch = True
+
+    args.result1.close()
+    args.result2.close()
```

Comment:
I think there is a code duplication here. Could you create a function to avoid it?